src/CircuitBreaker/src/HystrixBase/HystrixBaseOptions.cs

```diff
@@ -45,9 +45,9 @@ protected virtual long GetLong(string prefix, string key, string property, long
         protected virtual string GetString(string prefix, string key, string property, string globalDefault, string instanceDefaultFromCode)
         {
             var result = globalDefault;
-            result = (_dynamic != null) ? _dynamic.GetString($"{prefix}:default:{property}", result) : result; // dynamic global default
```

Comment:
While the parentheses in these expressions are all technically redundant, some of them (IMO) improve readability, such as these in ternary expressions. Maybe the better option for those cases would be to write them out in the longer form, but that is probably a separate discussion to be had

Reply:
Thanks for expressing this.

StyleCop, Roslyn, as well as Resharper/Rider provide automated handling of adding/removing parentheses in expressions to improve readability, of which Resharper/Rider provides the most flexible configuration:

![image](https://user-images.githubusercontent.com/104792814/172573599-6c756cf8-30f7-49ca-9ada-19588af6718f.png)

![image](https://user-images.githubusercontent.com/104792814/172573752-37a38baa-f69f-4822-9a03-91950c3515bf.png)

![image](https://user-images.githubusercontent.com/104792814/172573877-eefb28f3-b5ed-4007-895b-4689e218315f.png)

Above are the default settings, which make total sense to me: `a || b && c` is definitely confusing without parentheses, as well as bitwise operations, because their precedence rules are not well-known for most C# developers. Some folks believe that parenthesis around arithmetic terms improve readability as well, for example: `a + (b * c)`, which I find odd because we've all learned those precedence rules at elementary school.

Note that even Resharper/Rider does not support putting parentheses around terms in ternary expressions. I'm not aware of any analyzer for that, so I don't think there's anything we can do to enforce/normalize that, except to turn off automated rule-based analysis and handle it manually. Frankly I've never seen or heard the concern of parenthesis around ternary terms being raised before.

But once we configure a maximum line length, Resharper breaks ternary expressions as below, which you may like better:
```c#
var result = SomeLongExpressionThatExceedsMaximumLineLength
    ? ifTrueExpression
    : ifFalseExpression
```